dev/archery/archery/benchmark/core.py

```diff
@@ -0,0 +1,123 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+class Statistics:
+    def __init__(self, min, max, mean, stddev, q1, median, q3):
+        self.min = min
+        self.max = max
+        self.mean = mean
+        self.stddev = stddev
+        self.q1 = q1
+        self.median = median
+        self.q3 = q3
+
+    @staticmethod
+    def from_values(values):
+        sorted(values)
+        n = len(values)
+        mean = sum(values) / len(values)
+        sum_diff = sum([(val - mean)**2 for val in values])
+        stddev = (sum_diff / (n - 1))**0.5 if n > 1 else 0.0
```

Comment:
btw, since you're requiring Python 3 (I saw some f-strings), you should be aware that Python now has a simple [statistics module](https://docs.python.org/3/library/statistics.html) in its standard library.

Though it doesn't support arbitrary quantiles (there's an issue open for that : https://bugs.python.org/issue35775)

Reply:
Pandas sounds overkill for this, since you're dealing with arrays. Numpy would be enough.